Feature/update research

suinj8 opened the pull request:
research 관련 모든 컴포넌트 통일, 손 봤습니다

- 백엔드 코드와 변수명 통일

- 클래스형 함수형으로 변경

- new, detail 동시에 사용가능하도록 registerForm 이용, 수정

- 논문의 참여자 관련해서 수정했습니다
members를 따로 관리합니다. 
멤버의 값을 읽어 리스트의 값과 비교해 존재하면 배경색을 다르게합니다
한번더 클릭하면 배경색에 하얀색으로 바뀌며 input value에도 값이 빠집니다
즉 토글형식 같이 구현했습니다.
=> mui로 구현해보려고 했는데 typescript 문법을 써야하고 잘 이해도 되지않아서
기존의 mui 컴포넌트들로 직접 구현했습니다.

![image](https://user-images.githubusercontent.com/90179555/198879676-886cd1c5-59b0-4496-a712-faf68f245d72.png)
![image](https://user-images.githubusercontent.com/90179555/198879685-edf996d2-51e5-4396-abc2-393f7772dd0b.png)
이런 방식입니다.

suinj8 committed:
feat: update research

- axios 주소 변경(우선 데이터는 임시로 넣어놓음, 시각적으로 디자인 위해)
- 약간의 디자인 수정
- 객체 통일 + 비동기함수 통일
- 컴포넌트화 한 버튼 사용

suinj8 committed:
feat: update research

- 백엔드 코드의 변수명과 통일
- 클래스형식 함수형식으로 변경
- new와 detail 동시에 사용할 수 있도록 등록폼 생성
- Detail만 수정

suinj8 committed:
feat: update research

- registerform 수정
참여자 관련 로직수정
해당되는 멤버는 배경색을 다르게 주어 선택되었다는 것을 표시
다시 클릭하면 배경색이 빠지며 state에도 빠지고 input값에도 빠짐

- New 페이지들 registerform으로 합쳐 간소화

LimEunSang commented on src/admin/components/ResearchRegisterForm.js:
```diff
@@ -0,0 +1,362 @@
+    members.map((member) => (str += "," + member));
+    str = str.slice(1, str.length);
+    setStrMem(str);
+  }, [members]);
```
여기

LimEunSang commented on src/admin/components/ResearchRegisterForm.js:
```diff
@@ -0,0 +1,362 @@
+      ...research,
+      [e.target.name]: e.target.value,
+    });
+  };
```
여러 input을 하나의 함수로 관리하는 부분 (같은 방식으로 member 추후 수정)

LimEunSang reviewed (approved):
관련 함수는 모두 작성해놨지만 아직 backend 와의 연동은 확인 안해본거죠?
수고했습니다 LGTM!
(댓글로 코드 이해가 안가는 부분을 체크해두었습니다)

suinj8 committed:
feat: update researchField

researchField(연구분야)에 대해 백엔드 연동 끝냈습니다.
- 신규등록, 삭제, 변경기능 모두 구현
- 실제 실행결과 확인 완료
- 연동 끝남으로 더미데이터 삭제
- 각 주석 추가

suinj8 committed:
feat: update thesis

- 논문 전체 받아오기 백엔드 연동 완료
- 논문페이지 디자인 수정

euny0ung commented:
LGTM! 사소한 부분까지 신경쓰셨네요 수고하셨습니다 
research 페이지는 원래 제 담당이었으니 수정해야될 부분 저한테 고쳐달라고 하셔도 됩니다..!

suinj8 committed:
feat: update research

연동 임시 완료하였습니다.
바꾼 페이지가 많아 이미지 첨부 못한점 죄송합니다.
(실행시켜보시고 생성, 수정, 삭제, 자세히보기 확인 부탁드립니다)

- 데모, 프로젝트, 논문은 정상적으로 작동하는것 확인했습니다.
- 논문 부분 이슈 작성
- 데모 부분 이슈 작성
- api일치하지 않는 부분 issue에 업데이트 시킴
- 데모 디자인 변경
수정, 삭제버튼을 삭제함 => 어차피 자세히보기 페이지가 있는데
굳이 필요 없을거 같아서 삭제

suinj8 commented:
> LGTM! 사소한 부분까지 신경쓰셨네요 수고하셨습니다 research 페이지는 원래 제 담당이었으니 가능하면 저도 돕고싶은데 제가 할 일이 있다면 말씀해주세요 해볼게요!

일단 백엔드 연동까지는 저랑 은상이형이 끝내고 그 후에 프로젝트 전체적으로 다듬을 예정인데 그때 파트 나눠서
빠르게 진행합시다!

suinj8 committed:
feat: update research

연구 백엔드 연동 끝냈습니다.
- 논문의 참여자 구조 변경
- 데모 백엔드 수정 후 연동 확인
- 연구 전체 실행 확인

suinj8 commented:
@LimEunSang @euny0ung 
막힌 부분들까지 모두 해결 하였습니다.

논문에서 참여자들을 어떤식으로 보여줄까 고민하다가
(멤버 카테고리) (참여멤버 )
이런식으로 디자인을 변경하고
멤버 카테고리를 선택하면 해당 카테고리의 멤버를 불러와 참여멤버 리스트에 넣어줍니다.
선택되어 있는 멤버는 배경색을 다르게 하여 시각적으로 보이게 하였습니다.

이 외에 연구 부분은 모두 동작하는거 확인하였고 구현을 마쳤습니다.
그래도 혹시 모르니 각자 백엔드 코드 실행시켜서 확인해보시고
피드백 부탁드립니다.
(백엔드 코드는 develop 브랜치에서 실행시키셔야 합니다 - 어제 수정해야할 부분 민지님께 말씀드려서 수정함)

아래는 해당 구현부 사진입니다.
![image](https://user-images.githubusercontent.com/90179555/200104271-c482e765-e88b-4863-bace-b00dc909a21a.png)

suinj8 committed:
feat: bug fix

데모페이지 멤버를 읽는 버그가 발생하여 처리
사용안하는 import 제거
